asn1c/asn1c.c

```diff
@@ -179,6 +186,11 @@ main(int ac, char **av) {
         case 'S':
             skeletons_dir = optarg;
             break;
+        case 'D':
+            strncat(destdir, optarg, PATH_MAX - 2); /* leave room for possible trailing '/' */
+            if(destdir[strlen(destdir)-1] != '/')
+                strncat(destdir, "/", PATH_MAX - 1);
```

Comment:
The third argument to strncat doesn't do what you want it to do.

Reply:
I'm not sure I follow what I'm missing about strncat arg 3...
The following test program produces the expected output, and makes sure that destdir buffer does not overflow:
```c
#include <stdio.h>

#define MAX_CHARS 10
int main()
{
    const char *optarg = "123456789";
    char destdir[MAX_CHARS];
    destdir[0] = '\0';

    strncat(destdir, optarg, MAX_CHARS - 2); /* leave room for possible trailing '/' */
    if(destdir[strlen(destdir)-1] != '/')
        strncat(destdir, "/", MAX_CHARS - 1);
    
    printf("destdir='%s'\n", destdir);

    return 0;
}
```
Produces this output, that I expect:
```
destdir='12345678/'
```

Reply:
Ok. I see now... That line should be:
```c
strncat(destdir, "/", 1);
```
I will fix it.

Reply:
Yes. Though look at it, is pretty much useless this way. `strcat()` could do. `strncat()` is also ok though.

Reply:
Yes. I see. I've already left room for the trailing '/', so `strcat()` is sufficient. I will change it.